pkg/pool-utils/contracts/swapfees/SwapFeeController.sol

```diff
@@ -0,0 +1,142 @@
+ *
+ * In this implementation we ahve 4 levels of pools:
+ * 1. default pool - swap fee is equals maxSwapFeePercentage
+ * 2. regular pool - minSwapFeePercentageRegularPool <= swap_fee <= maxSwapFeePercentage
+ * 3. stableExotic pool - minSwapFeePercentageStableExoticPool <= swap_fee <= maxSwapFeePercentage
+ * 4. stableBlueChips pool - minSwapFeePercentageStableBCPool <= swap_fee <= maxSwapFeePercentage
+ *
+ * Vault's authorizer should grant permission to setPermission* functions.
+ */
+contract SwapFeeController is ISwapFeeController, Authentication {
+    // 1e18 corresponds to 1.0, or a 100% fee
+    uint256 private constant _MIN_SWAP_FEE_PERCENTAGE = 1e14; // 0.01%
+    uint256 private constant _MAX_SWAP_FEE_PERCENTAGE = 1e16; // 1% - this fits in 64 bits
+
+    IVault public immutable override vault;
+
+    uint256 public maxSwapFeePercentage;
+    uint256 public minSwapFeePercentageStableBCPool;
+    uint256 public minSwapFeePercentageStableExoticPool;
+    uint256 public minSwapFeePercentageRegularPool;
+
+    mapping(bytes32 => bool) private _stableBlueChipsPools;
+    mapping(bytes32 => bool) private _stableExoticPools;
+    mapping(bytes32 => bool) private _regularPools;
+
+    constructor(
+        IVault _vault,
+        uint256 _maxSwapFeePercentage,
+        uint256 _minSwapFeePercentageStableBCPool,
+        uint256 _minSwapFeePercentageStableExoticPool,
+        uint256 _minSwapFeePercentageRegularPool
+    ) Authentication(bytes32(uint256(address(this)))) {
+        require(address(_vault) != address(0), "Vault can't be empty");
+        _checkFeeBoundaries(_maxSwapFeePercentage);
+        _checkFeeBoundaries(_minSwapFeePercentageStableBCPool);
+        _checkFeeBoundaries(_minSwapFeePercentageStableExoticPool);
+        _checkFeeBoundaries(_minSwapFeePercentageRegularPool);
+
+        vault = _vault;
+        maxSwapFeePercentage = _maxSwapFeePercentage;
+        minSwapFeePercentageStableBCPool = _minSwapFeePercentageStableBCPool;
+        minSwapFeePercentageStableExoticPool = _minSwapFeePercentageStableExoticPool;
+        minSwapFeePercentageRegularPool = _minSwapFeePercentageRegularPool;
+
+        emit MaxSwapFeePercentageUpdated(maxSwapFeePercentage);
+        emit MinSwapFeePercentageStableBCPoolUpdated(minSwapFeePercentageStableBCPool);
+        emit MinSwapFeePercentageStableExoticPoolUpdated(minSwapFeePercentageStableExoticPool);
+        emit MinSwapFeePercentageRegularPoolUpdated(minSwapFeePercentageRegularPool);
+    }
+
+    function _checkFeeBoundaries(uint256 fee) private pure {
+        require(fee <= _MAX_SWAP_FEE_PERCENTAGE, "Fee should be within boundaries (big)");
+        require(fee >= _MIN_SWAP_FEE_PERCENTAGE, "Fee should be within boundaries (small)");
+    }
+
+    /**
+     * @notice Add/remove regular pool allowance
+     */
+    function setRegularPoolAllowance(bytes32 poolId, bool isAllow) external authenticate {
```

Comment:
make setters with array attribute for batch request

Reply:
Done